lib/licensed/sources/nuget.rb

```diff
@@ -0,0 +1,226 @@
+          return unless nuspec_contents
+
+          match = nuspec_contents.match LICENSE_URL_REGEX
+          if match && match[1]
+            url = original_url = match[1]
+
+            # Skip downloading if the URL contains a license identifier itself
+            if known_license = inspect_url_for_license(url)
+              return @nuspec_remote_license_file = KnownLicenseFile.new(known_license, { uri: url })
+            end
+
+            unless ignored_url?(url)
+              # Transform URLs that are known to return HTML but have a corresponding text-based URL
+              url = get_text_content_url(url)
+
+              # Attempt to fetch the license content
+              license_data = retrieve_license(url)
+              unless license_data.nil?
+                @nuspec_remote_license_file = Licensee::ProjectFiles::LicenseFile.new(license_data, { uri: original_url })
+              end
+            end
+          end
+        end
+
+        def get_text_content_url(url)
+          # Convert github file URLs to raw URLs
+          if match = url.match(/https?:\/\/(?:www\.)?github.com\/([^\/]+)\/([^\/]+)\/blob\/(.*)/i)
+            url = "https://github.com/#{match[1]}/#{match[2]}/raw/#{match[3]}"
+          else
+            url
+          end
+        end
+
+        def inspect_url_for_license(url)
+          if match = url.match(/https?:\/\/licenses.nuget.org\/(.*)/i)
+            Licensee::License.find(match[1])
+          elsif match = url.match(/https?:\/\/opensource.org\/licenses\/(.*)/i)
+            Licensee::License.find(match[1])
+          elsif match = url.match(/https?:\/\/(?:www\.)?apache.org\/licenses\/(.*?)(?:\.html|\.txt)?$/i)
+            Licensee::License.find(match[1].gsub("LICENSE", "Apache"))
+          end
+        end
+
+        def ignored_url?(url)
+          url == "https://aka.ms/deprecateLicenseUrl"
+        end
+
+        def retrieve_license(url, redirect_limit = 2)
+          return if redirect_limit == 0
+          begin
+            response = Net::HTTP.get_response(URI(url))
+            case response
+            when Net::HTTPSuccess     then response.body
+            when Net::HTTPRedirection then retrieve_license(response["location"], redirect_limit - 1)
+            end
+          rescue
+          end
+        end
+      end
+
+      def enabled?
+        nuget_config_exists(config.pwd)
+      end
+
+      def nuget_obj_root
+        config.dig("nuget", "obj_root") || config.pwd
+      end
+
+      def nuget_config_exists(root)
+        # Multiple supported casings: https://github.com/NuGet/Home/issues/1427
+        File.exist?(File.join(root, "nuget.config")) || File.exist?(File.join(root, "NuGet.config")) || File.exist?(File.join(root, "NuGet.Config"))
+      end
+
+      def enumerate_dependencies
+        packages.map do |key, package|
+          NuGetDependency.new(
+            name: "#{package["name"]} #{package["version"]}",
```

Comment:
How should multiple versions of the same dependency be handled? 

It looks like multiple dependencies with the same name (not metadata.name) result in only one file being emitted into the cache which makes status unhappy, so I made the dependency name / outputted file be “{name} {version}“, but no other source appears to do so.